core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java

```diff
@@ -917,6 +917,15 @@ public void testLoadTable() {
         .containsAll(properties.entrySet());
   }
 
+  @Test
+  public void testLoadTableWithNonExistingNamespace() {
+    TableIdentifier ident = TableIdentifier.of("non-existing", "tbl");
+    assertThat(catalog().tableExists(ident)).as("Table should not exist").isFalse();
+    assertThatThrownBy(() -> catalog().loadTable(ident))
+        .isInstanceOf(NoSuchTableException.class)
+        .hasMessageStartingWith("Table does not exist: %s", ident);
```

Comment:
I wonder if we can make more precise assertion than `.hasMessageStartingWith` just to avoid (unlikely) prefix match.

It seems `hasMessageEndingWith` is a better choice.

Reply:
there are catalog implementations that have additional details at the end of the message, hence we're checking the beginning of the error msg

Reply:
```
          throw new NoSuchTableException("Table does not exist: %s", identifier);
```
I wonder if punctuation (or a space) should be added at the end of message where the exception is thrown so that we eliminate the chance of false positive.